app/src/main/java/ar/com/wolox/android/example/ui/login/LoginFragment.java

```diff
@@ -20,6 +33,86 @@ public int layout() {
 
     @Override
     public void init() {
+        buttonLogin = getActivity().findViewById(R.id.button_login);
+        buttonSignup = getActivity().findViewById(R.id.button_signup);
+        textMail = getActivity().findViewById(R.id.text_mail);
+        textPass = getActivity().findViewById(R.id.text_pass);
+
+        obtenerCredenciales();
+
+        buttonLogin.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                String mail = textMail.getText().toString();
```

Comment:
Since all inside here is business logic, should be on the presenter. 
On MVP the `view` (a.k.a the fragment) inform events to the presenter, here could be something like `onLoginButtonClicked(user, password)`. Then the presenter will do all business logic and decide what should the `view` do. Like `view.showEmptyEmailError`,`view.showEmptyPasswordError`. 
As a note, the `view` should never know Android view stuff.